sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/token/generator/impl/AssistQueryAndPlainInsertColumnsTokenGenerator.java

```diff
@@ -52,9 +54,13 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
         Collection<InsertColumnsToken> result = new LinkedList<>();
         Optional<EncryptTable> encryptTable = getEncryptRule().findEncryptTable(sqlStatementContext.getTablesContext().getSingleTableName());
         Preconditions.checkState(encryptTable.isPresent());
+        Set<String> columnNames = new HashSet(((InsertStatement) sqlStatementContext.getSqlStatement()).getColumnNames());
         for (ColumnSegment each : ((InsertStatement) sqlStatementContext.getSqlStatement()).getColumns()) {
             List<String> columns = getColumns(encryptTable.get(), each);
             if (!columns.isEmpty()) {
+                if (columnNames.contains(columns.get(0))) {
+                    continue;
+                }
```

Comment:
I think move this logic into `getColumns` function will more reabable, is it?

Reply:
Yes, moved.